### PR TITLE
Fix some code defects

### PR DESCRIPTION
**Pull request by RyanUnderhill**
**Description**: Fix some code defects detected through automated tools.

3 defects detected, I'll put an explanation by each one in the diff.

### CODE REVIEW

File: orttraining/orttraining/test/graph/optimizer_graph_builder_test.cc [hunk 1]
```diff
@@ -196,7 +196,7 @@ static void TestOptimizerGraphBuilderWithInitialStates(OptimizerGraphConfig conf
   std::unordered_map<std::string, std::unordered_map<std::string, std::string>> opt_initializer_names_map;
   ASSERT_STATUS_OK(optimizer_graph_builder.Build(graph, opt_initializer_names_map, opt_graph_outputs));
 
-  const ONNX_NAMESPACE::TensorProto* tensor;
+  const ONNX_NAMESPACE::TensorProto* tensor{};
```
Comment (RyanUnderhill) [inline orttraining/orttraining/test/graph/optimizer_graph_builder_test.cc:199]: Access of possibly uninitialized memory.

File: orttraining/orttraining/training_ops/cuda/nn/conv_grad.cc [hunk 2]
```diff
@@ -222,8 +222,8 @@ class AlgoIterator {
 
   Status TryAll(const CUDAExecutionProvider* provider, std::function<Status(const T_Perf& perf)> f) {
     auto& cache = AlgoSearch<T_Perf>::Cache();
-    T_Perf algo_perf;
-    if (cache.Find(args_.params, &algo_perf) && f(algo_perf) == Status::OK()) {
+    
+    if (T_Perf algo_perf; cache.Find(args_.params, &algo_perf) && f(algo_perf) == Status::OK()) {
```
Comment (RyanUnderhill) [inline orttraining/orttraining/training_ops/cuda/nn/conv_grad.cc:226]: Was hidden by a deeper algo_perf variable in the next scope. This keeps it within the scope of the if.

File: orttraining/orttraining/core/graph/graph_augmenter.h [hunk 3]
```diff
@@ -182,7 +182,7 @@ class GraphAugmenter {
     }
 
     TypeProto* CopyTypeProto(const NodeArg* node_arg) {
-      ORT_ENFORCE(node_arg != nullptr, "During CopyTypeProto, ", node_arg->Name(), "'s node_arg is null.");
+      ORT_ENFORCE(node_arg != nullptr, "During CopyTypeProto, node_arg is null.");
```
[the remark this answers is not in the document]
Comment (yuslepukhin) [reply orttraining/orttraining/core/graph/graph_augmenter.h:185]: Really bad.

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**